datadog-opentelemetry/src/lib.rs

```diff
@@ -477,3 +513,54 @@ pub fn make_test_tracer(shared_config: Arc<Config>) -> (SdkTracerProvider, Datad
         None,
     )
 }
+
+use opentelemetry_sdk::metrics::SdkMeterProvider;
+
+/// Builder for Datadog Metrics with OTLP transport
+pub struct DatadogMetricsBuilder {
+    config: Option<Config>,
+    resource: Option<Resource>,
+    export_interval: Option<std::time::Duration>,
+}
+
+impl DatadogMetricsBuilder {
+    pub fn with_config(mut self, config: Config) -> Self {
+        self.config = Some(config);
+        self
+    }
+
+    pub fn with_resource(mut self, resource: Resource) -> Self {
+        self.resource = Some(resource);
+        self
+    }
+
+    pub fn with_export_interval(mut self, interval: std::time::Duration) -> Self {
+        self.export_interval = Some(interval);
+        self
+    }
+
+    pub fn init(self) -> Result<SdkMeterProvider, String> {
+        let (meter_provider, _) = self.init_local()?;
+        opentelemetry::global::set_meter_provider(meter_provider.clone());
+        Ok(meter_provider)
+    }
+
+    pub fn init_local(self) -> Result<(SdkMeterProvider, ()), String> {
+        let config = self.config.unwrap_or_else(|| Config::builder().build());
+        let meter_provider = crate::metrics_reader::create_meter_provider(
+            Arc::new(config),
+            self.resource,
+            self.export_interval,
+        )?;
+        Ok((meter_provider, ()))
+    }
+}
```

Comment:
I don't think those methods should ever return errors.
I'd prefer if we returned an no-op SdkMeterProvider (with no exporter) and logged a scary log, rather than customers having to think about what if the meter provider is missing.

Also in general, return `Result<T, String>` feels weird... I don't think it's completely invalid, but I'd prefer if you used `anyhow::Result<T>` instead.
The error is also typed erased and can be a just a string, but it also captures the backtrace and you can wrap errors more efficiently.